corelib-storage/src/main/java/eu/europeana/corelib/solr/derived/AttributionConverter.java

```diff
@@ -96,22 +96,24 @@ private void checkEuropeanaAggregration(Attribution attribution, EuropeanaAggreg
     // set Attribution with creator values. Checks for URI and adds their labels from Agents. Adds the NonURI values too
     public void checkCreatorLabel(Attribution attribution, List<Agent> agents, Map<String, List<String>> creatorMap) {
         Map<String, List<String>> finalMap = new HashMap<>();
-        for (Map.Entry<String, List<String>> creator : creatorMap.entrySet()) {
-            List<String> creatorValues = new ArrayList<>();
-            for (String value : creator.getValue()) {
-                if (EuropeanaUriUtils.isUri(value)) {
-                    for (Agent agent : agents) {
-                        if (StringUtils.equals(value, agent.getAbout())) {
-                            creatorValues.addAll(getCreatorFromAgent(agent));
+        if(!creatorMap.isEmpty() && !agents.isEmpty()) {
+            for (Map.Entry<String, List<String>> creator : creatorMap.entrySet()) {
+                List<String> creatorValues = new ArrayList<>();
+                for (String value : creator.getValue()) {
+                    if (EuropeanaUriUtils.isUri(value)) {
+                        for (Agent agent : agents) {
+                            if (StringUtils.equals(value, agent.getAbout())) {
+                                creatorValues.addAll(getCreatorFromAgent(agent));
+                            }
```

Comment:
This code is 6 levels deep. It should be refactored to use submethods.

Reply:
done